### ISSUE TITLE
Initialization fails when upgrading to the 2022.01.1 docker image

### ISSUE THREAD

**Issue by Neurrone** [pi-hole/docker-pi-hole#974]
### Versions

- Pi-hole: 5.8 
- AdminLTE: 5.10 
- FTL: 5.13 

### Platform

- OS and version: DSM7 on a Synology DS918+ 

- Platform: Docker using tag 2022.01.1

### Expected behavior

Pihole initializes successfully

### Actual behavior / bug

Pihole startup fails with this error:

```
pihole         | dnsmasq: cannot access directory /etc/dnsmasq.d: Permission denied                                                  
pihole         | ::: Testing pihole-FTL DNS: [cont-init.d] 20-start.sh: exited 1.                                                    
```

Reverting back to the previous docker image tag, 2021.12.1 fixed this issue for me.

### Steps to reproduce

1. Use the container tagged 2022.01.1 in Docker, upgrading from a previous version. Unsure if this can be reproduced with fresh installs.
2. `docker-compose up`

### Additional context

_Add any other context about the problem here._

**Comment by PromoFaux** [comment]
What does your compose file / run script look like, please?

**Comment by Neurrone** [comment]
```
version: '2' # required for macvlan, which isnt supported in v3

services:
  cloudflared:
    container_name: cloudflared
    restart: unless-stopped
    image: cloudflare/cloudflared:2022.1.0
    command: proxy-dns
    environment:
      # use cloudflare and quad 9's DNS resolvers
      - "TUNNEL_DNS_UPSTREAM=https://1.1.1.1/dns-query,https://1.0.0.1/dns-query,https://9.9.9.9/dns-query,https://149.112.112.9/dns-query"
      # Listen on an unprivileged port
      - "TUNNEL_DNS_PORT=5053"
      # Listen on all interfaces
      - "TUNNEL_DNS_ADDRESS=0.0.0.0"
    # Attach cloudflared only to the private network
    networks:
      internal:
        ipv4_address: 172.30.9.2
  pihole:
    # image: pihole/pihole:2022.01.1
    image: pihole/pihole:2021.12.1
    depends_on:
      - cloudflared
    container_name: pihole
    hostname: pihole
    networks:
      internal:
        ipv4_address: 172.30.9.3
      pihole_network:
        ipv4_address: 192.168.2.200
    # For DHCP it is recommended to remove these ports and instead add: network_mode: "host"
    # ports:
      # - "53:53/tcp"
      # - "53:53/udp"
      # - "67:67/udp"
      # - "80:80/tcp"
      # - "443:443/tcp"
    environment:
      DNSMASQ_LISTENING: local
      ServerIP: 192.168.2.200
      TZ: Asia/Singapore
      WEBPASSWORD: ...
      DNS1: "172.30.9.2#5053"
      DNS2: "no"
    restart: unless-stopped
    volumes:
      - './etc-pihole/:/etc/pihole/'
      - './etc-dnsmasq.d/:/etc/dnsmasq.d/'
      - './var-log/pihole.log:/var/log/pihole.log'
    # Recommended but not required (DHCP needs NET_ADMIN)
    #   https://github.com/pi-hole/docker-pi-hole#note-on-capabilities
    # cap_add:
      # - NET_ADMIN

networks:
  pihole_network:
    driver: macvlan
    driver_opts:
      parent: eth0
    ipam:
      config:
        - subnet: 192.168.2.0/24
          gateway: 192.168.2.1
          ip_range: 192.168.2.192/28
  # for communication with cloudflared
  internal:
    ipam:
      config:
        - subnet: 172.30.9.0/29
        - ```

**Comment by PromoFaux** [comment]
> Reverting back to the previous docker image tag, 2021.12.1 fixed this issue for me.

Previous tag is actually [`2022.12`](https://github.com/pi-hole/docker-pi-hole/releases/tag/2022.01), and that is where the [change that is affecting you](https://github.com/pi-hole/docker-pi-hole/issues/963) happened.

TL;DR - check the permissions on your bind mounts - or add `DNSMASQ_USER: root` to your environment - it pays to check the release notes before updating :)

**Comment by Neurrone** [comment]
Thanks, that fixed my issue 🙏
